mongo/integration/sessions_mongocryptd_prose_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/event"
+	"go.mongodb.org/mongo-driver/internal/assert"
+	"go.mongodb.org/mongo-driver/internal/require"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+type mongocryptdProcess struct {
+	cmd *exec.Cmd
+}
+
+// start will start a mongocryptd server in the background on the OS.
+func (p *mongocryptdProcess) start(port int) error {
+	args := []string{
+		"mongocryptd",
+		"--port", strconv.Itoa(port),
+	}
+
+	p.cmd = exec.Command(args[0], args[1:]...) //nolint:gosec
+	p.cmd.Stderr = p.cmd.Stdout
+
+	return p.cmd.Start()
+}
+
+// close will kill the underlying process on the command.
+func (p *mongocryptdProcess) close() error {
+	if p.cmd.Process == nil {
+		return nil
+	}
+
+	if err := p.cmd.Process.Kill(); err != nil {
+		return fmt.Errorf("failed to terminate mongocryptd process: %w", err)
+	}
+
+	// Release instead of wait to avoid blocking in CI.
+	err := p.cmd.Process.Release()
+	if err != nil {
+		return fmt.Errorf("failed to release mongocryptd: %w", err)
+	}
+
+	return nil
+}
+
+func TestSessionsMongocryptdProse(t *testing.T) {
+	const mongocryptdPort = 27022
+
+	// Monitor the lsid value on commands. If an operation run in any
+	// subtests contains an lsid, then the Go Driver wire message
+	// construction has incorrectly interpreted that
+	// LogicalSessionTimeoutMinutes was returned by the server on handshake.
+	cmdMonitor := &event.CommandMonitor{
+		Started: func(_ context.Context, evt *event.CommandStartedEvent) {
+			_, err := evt.Command.LookupErr("lsid")
+			if !errors.Is(err, bsoncore.ErrElementNotFound) {
+				require.NoError(t, err, "expected error to be nil, got %v", err)
+			}
+		},
+	}
+
+	uri := &url.URL{
+		Scheme: "mongodb",
+		Host:   net.JoinHostPort("localhost", strconv.Itoa(mongocryptdPort)),
+	}
+
+	mtOpts := mtest.NewOptions().
+		MinServerVersion("5.0").
```

Comment:
```suggestion
		MinServerVersion("4.2").
```

mongocryptd is available in server 4.2+. I expect there is no additional requirement for 5.0+ server.